src/mesh_analysis.rs

```diff
@@ -327,11 +219,11 @@ mod tests {
         let oriented_edges_non_manifold_check: Vec<_> =
             non_manifold_edges(&edge_valency_map).collect();
 
-        for o_e in &oriented_edges_non_manifold_check {
-            assert!(oriented_edges_non_manifold_correct.iter().any(|e| e == o_e));
+        for o_e in &oriented_edges_non_manifold_correct {
+            assert!(oriented_edges_non_manifold_check.iter().any(|e| e == o_e));
         }
 
-        assert_eq!(oriented_edges_non_manifold_check.len(), 1);
+        assert_eq!(oriented_edges_non_manifold_check.len(), 3);
```

Comment:
because 1 was a wrong assumption. since we're asking for oriented edges, they are in fact 3 (therefore the edge is non-manifold). there is, however, just one edge, appearing 3 times as unoriented or twice plus one as oriented

Reply:
kk, thought so, thanks!